jina/executors/indexers/keyvalue.py

```diff
@@ -75,6 +77,34 @@ def query(self, key: int) -> Optional[bytes]:
             with mmap.mmap(self.query_handler.body, offset=p, length=l) as m:
                 return m[r:]
 
+    def update(self, keys: Iterator[int], values: Iterator[bytes], *args, **kwargs):
+        # check: hard fail (raises) on key not found
+        missed = []
+        for key in keys:
+            if self.query_handler.header.get(key) is None:
+                missed.append(key)
+        if missed:
+            raise KeyError(f'Key(s) {missed} were not found in {self.save_abspath}')
```

Comment:
I guess you don't need to loop over all the keys and raise the error.
```python
for key in keys:
    if not self.query_handler.header.get(key):
        raise KeyError('...')
```

Reply:
Nice point. True, all missed update-keys are collected in memory. As a tradeoff we get have a better error handling. 
Actually, we have all keys in memory already (ReadHandler.header). Since keys are expected to be much smaller than values, we decided to accept the tradeoff.

Reply:
We want to log them all in one go